Makefile

```diff
@@ -520,12 +502,8 @@ test-admin:
 test-enterprise:
 	go test -v ./src/server/enterprise/server -count 1 -timeout $(TIMEOUT)
 
-# TODO This is not very robust -- it doesn't work when the PACHD_ADDRESS host isn't an IPv4 address
-PACHD_HOST := $(word 1,$(subst :, ,$(PACHD_ADDRESS)))
-PACHD_PORT := $(word 2,$(subst :, ,$(PACHD_ADDRESS)))
-
 test-tls:
-	./etc/testing/test_tls.sh
+	./etc/testing/test_tls.sh --host=$(PACHD_HOST) --port=$(PACHD_PORT)
```

Comment:
Yeah, you'd just have to set `PACHD_HOST`/`PACHD_PORT`, or call `./etc/testing/test_tls.sh` with `--host` and `--port`. I can still work on this improvement though.

Reply:
Oh actually, assuming jq is on the system, we can do this now: `pachcctl config get context foo | jq -r .pachd_address`

Reply:
I actually prefer the approach of adding a `pachctl address` command (though, again, doesn't have to be in this PR) because of the default-port logic in `client.go`.

I actually initially wanted to dispute the approach of adding a default port, even though that's what `client.go` did for `PACHD_ADDRESS`—I felt that `pachctl` should alter neither `PACHD_ADDRESS` nor the config address—specifically so that tests and other code could determine `pachctl`'s address  by reading the config themselves (e.g. with `jq` like above). The problem is that then the tests also have to add the default port if it's missing, and in general both the logic in `client.go` and the bash that runs our tests have to be kept in sync.

I now believe that code besides `pachctl` just shouldn't read the config. To get `pachctl`'s address, the test code should be able to ask `pachctl` what it's address it, and `pachctl` should read the config.

Obviously let me know if you disagree, but that was my line of thought

Reply:
OK, I rewrote it to do that; let me know what you think

Reply:
(for posterity: see discussion at the end of the PR, as well as https://github.com/pachyderm/pachyderm/pull/4208. The resolution was *not* to create a `pachctl address` command, but instead to canonicalize `pachd_address` and other config fields at config write time. This way, other tools can read the `pachctl` config directly and get a canonicalized address)